scripts/gen-deps-list.js

```diff
@@ -1,110 +1,101 @@
+  const content = await fs.readFile(changelogConfig.path)
+  const changelogRegex = new RegExp(
+    `${escapeRegExp(changelogConfig.startTag)}(.*)${escapeRegExp(changelogConfig.endTag)}`,
+    's'
+  )
+  const block = changelogRegex.exec(content)?.[1].trim()
+
+  if (block === undefined) {
+    throw new Error(`Could not find changelog block in ${changelogConfig.path}`)
   }
 
-  const packages = await getPackages(true)
-  const rootPackage = packages.find(pkg => pkg.name === rootPackageName)
+  const lines = block.split(/\s*\r?\n\s*/).map(line => {
+    const items = trim(line, '- ').split(/\s+/)
 
-  if (!rootPackage) {
-    program.showHelpAfterError(false).error(`error: Package ${rootPackageName} not found`)
-  }
+    if (items.length !== 2) {
+      throw new Error(`Invalid line: "${line}"`)
+    }
 
-  dependencyTree.add(rootPackage.name)
-
-  /**
-   * Recursively add dependencies to the dependency tree
-   *
-   * @param {string} handledPackageName The name of the package to handle
-   * @param {string} handledPackageNextVersion The next version of the package to handle
-   */
-  function handlePackage(handledPackageName, handledPackageNextVersion) {
-    packages.forEach(({ package: { name, version, dependencies, optionalDependencies, peerDependencies } }) => {
-      let releaseWeight
-
-      if (
-        shouldPackageBeReleased(
-          handledPackageName,
-          { ...dependencies, ...optionalDependencies },
-          handledPackageNextVersion
-        )
-      ) {
-        releaseWeight = RELEASE_WEIGHT.PATCH
-      } else if (shouldPackageBeReleased(handledPackageName, peerDependencies || {}, handledPackageNextVersion)) {
-        releaseWeight = versionToReleaseWeight(version)
-      }
-
-      if (releaseWeight !== undefined) {
-        registerPackageToRelease(name, releaseWeight)
-        dependencyTree.add(name, handledPackageName)
-        handlePackage(name, getNextVersion(version, releaseWeight))
-      }
-    })
-  }
+    return items
+  })
+
+  for (const [name, releaseType] of lines) {
+    if (!['patch', 'minor', 'major'].includes(releaseType)) {
+      throw new Error(`Package "${name}" has invalid release type: "${releaseType}"`)
+    }
+
+    const rootPackage = allPackages.find(pkg => pkg.name === name)
 
-  handlePackage(rootPackage.name, getNextVersion(rootPackage.version, rootReleaseWeight))
+    if (!rootPackage) {
+      throw new Error(`Package "${name}" does not exist`)
+    }
+
+    const rootReleaseWeight = releaseTypeToWeight(releaseType)
+    registerPackageToRelease(name, rootReleaseWeight)
+    dependencyTree.add(rootPackage.name)
+
+    handlePackageDependencies(rootPackage.name, getNextVersion(rootPackage.version, rootReleaseWeight))
+  }
 
   const outputLines = dependencyTree.resolve().map(dependencyName => {
     const releaseTypeName = RELEASE_TYPE[packagesToRelease.get(dependencyName)].toLocaleLowerCase()
     return `- ${dependencyName} ${releaseTypeName}`
```

Comment:
I think that would be helpful to have 2 outputs: one to run `bump-pkg` and one for `CHANGELOG.md`, maybe with the following format:

```
Commands to execute:

./scripts/bump-pkg <name> <release type>

### Released packages

- <name> <version>
```

@fbeauchamp What do you think?
```

Reply:
I think I will only use it during the release , to update changelog

Reply:
The goal of this script is to compute which packages need to be released (hence the list of commands).

But I think it can also help us update the changelog, that's what I'm proposing here :slightly_smiling_face:

Reply:
I'm not sure what is exactly expected here.

If the current version of the script is OK to make a release, is it possible to create a new kanban feature card for additional changes?

Reply:
Sure, but the changes I'm proposing are very minor.

And I'm also wondering if the current output is the best for the current goal.

Reply:
@julien-f Thanks for the call. I clearly understand the needs now (i thought the changes was really more important).

No need for a new kanban card then. I'll do the change here.